### PR TITLE
Convert EditRetryDialog to Typescript

### PR DESCRIPTION
**Pull request by TravisNickels**

### CODE REVIEW

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+import type Header from "@/resources/Header";
+
+interface HeaderWithEditing extends Header {
+  isLocked: boolean;
+  isSensitive: boolean;
+  isMarkedAsRemoved: boolean;
+  isChanged: boolean;
+}
 
 const emit = defineEmits(["cancel", "retried"]);
```
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:16]: You can type emits, see https://vuejs.org/guide/typescript/composition-api.html#typing-component-emits

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 2]
```diff
@@ -178,16 +186,16 @@ onMounted(() => {
                       </div>
                       <table class="table" v-if="panel === 1">
                         <tbody>
-                          <tr class="interactiveList" v-for="(header, index) in localMessage.headers" :key="index">
+                          <tr class="interactiveList" v-for="(header, index) in localMessage?.headers" :key="index">
```
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:189]:
```suggestion
                          <tr class="interactiveList" v-for="(header, index) in localMessage?.headers" :key="header.key">
```

**Review by cquirosj — APPROVED** [review]

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
-let origMessageBody = undefined;
+let origMessageBody: string;
 
 const showEditAndRetryConfirmation = ref(false);
 const showCancelConfirmation = ref(false);
 const showEditRetryGenericError = ref(false);
 
-const id = computed(() => settings.id);
-const messageBody = computed(() => settings.message.messageBody);
+const id = computed(() => props.id);
```
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:35]: This is unnecessary

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 4]
```diff
@@ -1,26 +1,39 @@
...
+const messageBody = computed(() => props.message.messageBody);
```
Comment (johnsimons) [inline src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:36]: This also seems unecessary?

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 5]
```diff
@@ -60,47 +73,43 @@ function resetBodyChanges() {
...
-function findHeadersByKey(key) {
-  return localMessage.value.headers.find((header) => header.key === key);
+function findHeadersByKey(key: string) {
+  return localMessage.value.headers.find((header: HeaderWithEditing) => header.key === key);
 }
 
 function getContentType() {
   const header = findHeadersByKey("NServiceBus.ContentType");
-  return header?.value;
+  return header.value;
```
Comment (PhilBastian) [inline src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:82]: why does this not require a ? anymore? find() can return undefined

File: src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue [hunk 6]
```diff
@@ -60,47 +73,43 @@ function resetBodyChanges() {
...
+  return intent.value;
 }
 
 function removeHeadersMarkedAsRemoved() {
-  localMessage.value.headers.forEach((header, index, object) => {
-    if (header.isMarkedAsRemoved) {
-      object.splice(index, 1);
-    }
-  });
+  localMessage.value.headers = localMessage.value.headers.filter((header: HeaderWithEditing) => !header.isMarkedAsRemoved);
```
Comment (PhilBastian) [inline src/ServicePulse.Host/vue/src/components/failedmessages/EditRetryDialog.vue:95]: i do love the... creativeness... of the old solution 🤣

**Comment by johnsimons** [comment]
it seems this is already merged!

**Pull request merged**